pkg/transform/oauth/oauth.go

```diff
@@ -144,10 +148,15 @@ func Translate(identityProviders []IdentityProvider) (*CRD, []secrets.Secret, er
 			continue
 		}
 
+		// Check if config map is not empty
+		if caConfigMap != (configmaps.ConfigMap{}) {
```

Comment:
I would be careful with this kind of statement. This probably does what we're asking it to do here, but testing for "Equality" on two types can be much more complex than it appears to be on the surface (deep vs. shallow checks).

I think the better golike/kubelike way of representing the absence of a ConfigMap is to have them return nil, and not an empty struct. This pattern is used all over go, as an example, with returning errors from functions. It's also done all over kubernetes to represent an optional type, used to distinguish the absence of something vs. the presence of something that exists, but is empty.